lib/elastic_apm/metrics.rb

```diff
@@ -51,14 +56,19 @@ def start
 
         # Only set the @sets once, in case we stop
         # and start again.
-        @sets ||= {
-          system: CpuMemSet,
-          vm: VMSet,
-          breakdown: BreakdownSet,
-          transaction: TransactionSet
-        }.each_with_object({}) do |(key, kls), sets|
-          debug "Adding metrics collector '#{kls}'"
-          sets[key] = kls.new(config)
+        if @sets.nil?
+          sets = {
+            system: CpuMemSet,
+            vm: VMSet,
+            breakdown: BreakdownSet,
+            transaction: TransactionSet
+          }
+          sets[:jvm] = JVMSet if defined?(JVMSet)
```

Comment:
I've added this small change in my local environment to make it easier to tell if this conditional worked. Seems useful.
```suggestion
          if defined?(JVMSet)
            debug "Enabling JVM metrics collection"
            sets[:jvm] = JVMSet
          end
```